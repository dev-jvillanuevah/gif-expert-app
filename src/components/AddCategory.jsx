import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState('');

  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmitValue = (event) => {
    event.preventDefault();
    const value = inputValue.trim();
    if (value.length < 1) return;

    onNewCategory(value);
    setInputValue('');
  };

  return (
    <form onSubmit={onSubmitValue} aria-label="form">
      <input
        type="text"
        placeholder="Search gifs"
        value={inputValue}
        onChange={onInputChange}
        //   onChange={event => onInputChange(event)} I send the first prop I can do it like the one on top
      />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
