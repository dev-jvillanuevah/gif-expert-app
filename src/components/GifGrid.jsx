import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifItem } from './GifItem';

export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      {isLoading && <h2>Loading...</h2>}{' '}
      {/* If isLoading is true it will show the second argument (<h2>Loading...</h2>) */}
      <div className="card-grid">
        {' '}
        {/* This will now show until we have the images loaded */}
        {images.map((image) => (
          <GifItem key={image.id} {...image} />
        ))}
      </div>
    </>
  );
};
