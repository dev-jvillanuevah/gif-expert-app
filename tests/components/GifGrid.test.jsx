const { render, screen } = require('@testing-library/react');
import { GifGrid } from '../../src/components';
import { useFetchGifs } from '../../src/hooks/useFetchGifs';
jest.mock('../../src/hooks/useFetchGifs');

describe('tests on GifGrid', () => {
  const category = 'Dragon Ball Z';
  test('should show the loading initially', () => {
    useFetchGifs.mockReturnValue({
      images: [],
      isLoading: true,
    });
    render(<GifGrid category={category} />);
    expect(screen.getByText('Loading...'));
    expect(screen.getByText(category));
  });

  test('should show items when the images from useFetchGifs are loaded', () => {
    const gifs = [
      {
        id: 'ABC',
        title: 'Goku',
        url: 'https://localhost/goku.jpg',
      },
      {
        id: '123',
        title: 'Vegeta',
        url: 'https://localhost/vegeta.jpg',
      },
    ];
    useFetchGifs.mockReturnValue({
      images: gifs,
      isLoading: false,
    });
    render(<GifGrid category={category} />);
    expect(screen.getAllByRole('img').length).toBe(2);
  });
});
