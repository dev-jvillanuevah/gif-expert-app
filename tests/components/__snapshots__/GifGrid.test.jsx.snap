// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on GifGrid should match with the snapshot 1`] = `
<div>
  <h3>
    Dragon Ball Z
  </h3>
  <h2>
    Loading...
  </h2>
   
  <div
    class="card-grid"
  >
     
  </div>
</div>
`;
